src/services/daam/backend/submitApplication.ts

```diff
@@ -0,0 +1,20 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { ErrorResponse, ValidationWarnings } from '@/types/api.types';
+import { ExtendedSession } from '@/utils/auth';
+import { decrypt } from '@/utils/encryption';
+
+export function makeSubmitApplication(daamUrl: string) {
+  return async (applicationId: string, session: ExtendedSession): Promise<void | ErrorResponse | ValidationWarnings> => {
+    const response = await fetch(`${daamUrl}/api/v1/applications/${applicationId}/submit`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${decrypt(session.access_token)}`,
+      },
+    });
+    if (!response.ok) return response.json();
```

Comment:
i suggest to directly return response and do the error handling in routes/applications file because based on the error there you could return appropriate response as done here
https://github.com/GenomicDataInfrastructure/gdi-userportal-frontend/pull/196/files

Reply:
The problem is that I am getting an error `unhandledRejection: SyntaxError: Unexpected end of JSON input` when getting a success (204 no content). That's why I put the condition here

Reply:
ah because in this case there is no json body. You need to check what happens when an backend exception is thrown. Does it get catched here properly?
https://github.com/GenomicDataInfrastructure/gdi-userportal-frontend/pull/194/files#diff-27665da4eb33be756d33100feda86f6d6ffcc491ce925c251d043a64a8a3039dR23